MobileBuy/buy/src/main/java/com/shopify/buy3/GraphCall.java

```diff
@@ -30,36 +30,129 @@
 
 import com.shopify.graphql.support.AbstractResponse;
 
+/**
+ * <p> {@code GraphCall} is an abstraction for a {@code GraphQL} request that has been prepared for execution. As this class represents a
+ * single request/response pair, it can't be executed twice. A call can be canceled.</p>
+ * <p>In order to execute the request again, use {@link GraphCall#clone()} method which creates a new identical call object.</p>
+ *
+ * @param <T> type of the {@link AbstractResponse} response this call returns
+ */
 public interface GraphCall<T extends AbstractResponse<T>> {
 
+  /**
+   * Cancels the call, if possible.
+   */
   void cancel();
 
+  /**
+   * Checks if this call has been canceled.
+   *
+   * @return {@code true} if this call has been canceled otherwise {@code false}
+   */
   boolean isCanceled();
 
+  /**
+   * Creates a new, identical call to this one which can be enqueued or executed even if this call has already been executed or canceled.
+   *
+   * @return new cloned call
+   */
   @NonNull GraphCall<T> clone();
 
+  /**
+   * Sends the {@code GraphQL} request immediately and blocks until the response can be processed or is an error.
+   *
+   * @return {@code GraphQL} response
+   * @throws IllegalStateException when the call has already been executed
+   * @throws GraphError            if the request could not be executed due to a cancellation, a timeout, network failure, parsing error
```

Comment:
Yes so if OkHttp request has been canceled then it will throw IOException:
https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/RealCall.java#L70
https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/RealCall.java#L138